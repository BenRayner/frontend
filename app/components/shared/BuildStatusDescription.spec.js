/* global describe, it, expect, jest */
import moment from 'moment';
import MockDate from 'mockdate';
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

describe('BuildStatusDescription', () => {
  it(`calls through to \`friendlyRelativeTime\` and \`buildStatus\``, () => {
    MockDate.set(moment.parseZone("2016-10-10T00:21:50+1100"));

    const BUILD_STATUS_FIXTURE = {
      prefix: 'tested',
      timeValue: moment.parseZone("2016-10-09T23:58:05+1100")
    };

    const fauxFriendlyRelativeTime = jest.fn(() => 'at some point');
    const fauxBuildStatusDescription = jest.fn(() => BUILD_STATUS_FIXTURE);

    jest.mock('../../lib/friendlyRelativeTime', () => fauxFriendlyRelativeTime);
    jest.mock('../../lib/builds', () => ({ buildStatus: fauxBuildStatusDescription }));

    const BuildStatusDescription = require('./BuildStatusDescription').default;

    const build = {
      state: 'testing',
      canceledAt: new Date(1475643467000),
      createdAt: new Date(1475643466920),
      finishedAt: new Date(1475643467000)
    };

    const component = ReactTestRenderer.create(
      <BuildStatusDescription build={build} updateFrequency={0} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(fauxFriendlyRelativeTime).toHaveBeenCalledTimes(2);
    expect(fauxFriendlyRelativeTime).toHaveBeenCalledWith(BUILD_STATUS_FIXTURE.timeValue);
    expect(fauxBuildStatusDescription).toHaveBeenCalledTimes(1);
    expect(fauxBuildStatusDescription).toHaveBeenCalledWith(build);
    MockDate.reset();
  });
});
