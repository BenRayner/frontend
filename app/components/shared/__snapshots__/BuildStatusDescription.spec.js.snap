exports[`BuildStatusDescription calls through to \`friendlyRelativeTime\` and \`buildStatus\` 1`] = `
<span>
  tested
   
  <time
    dateTime={"2016-10-09T12:58:05.000Z"}
    title="Sunday, October 9, 2016 11:58 PM">
    at some point
  </time>
</span>
`;
